Proposal for an AI policy (#73)
YOURLS/.github: AI_POLICY.md

Reviewer: Should be named `AI_POLICY.md`, no?

Author: LOL yes, sorry. Fat fingers, late writing, and mixing up English with French :D

# AI Usage Policy

The YOURLS project and organization have strict rules regarding AI usage.

- **All AI usage in any form must be disclosed.** You must state

>   the tool you used (e.g. Claude Code) along with the extent to which the work

Reviewer 2: ```suggestion
  the AI model you used along with the extent to which the work
```
* The model gives more hint on the dry quality
* I'd prefer we don't mention any specific stuff that might die in less than a year.

- **The human-in-the-loop must fully understand all code.** If you can't explain
  without AI what your changes do and how they interact with code, please do not

>   contribute to this project.

Reviewer: I should've nitpicked this one before, sorry.

```suggestion
  without AI what your changes do and how they interact with existing code, please
  do not contribute to this project.
```

- **Any content generated with AI must have been reviewed _and edited_
  by a human before submission.** AI is very good at being overly verbose and
  including noise that distracts from the main point.

- **The human-in-the-loop must communicate with maintainers.** This follows from
  the preceding points: AI models are verbose and unfocused. Don't have your AI
  tool describe to us what you've found or changed; say it in your own words.

- **Pull Requests must be focused.** If you want to submit a pull request that fixes a

>   problem, enhances or introduce a new feature, make sure your code does just

Reviewer: ```suggestion
  problem, enhances or introduces a new feature, make sure your patch does only
```

  problem, enhances or introduce a new feature, make sure your code does just
  that. PRs that change whitespace, reword comments, or contain multiple

>   unrelated sets of changes will be rejected.

Reviewer 2: ```suggestion
- **Pull Requests must be focused.** If you want to submit a pull request that fixes a
  problem, enhances or introduce a new feature, make sure your code does just
  that. Pull requests that change whitespace, reword comments, or contain multiple
  unrelated sets of changes will be rejected.
```

Just to be sure "PR" is understood correctly 🫣

- **Pull Requests must originate from a real human need.** A valid pull request addresses
  a bug you actually encountered, or a feature you genuinely need. We do not
  accept pull requests where the origin is an AI agent that scanned the codebase and

>   decided something could be "improved" -- refactored, renamed, reworded, or

Reviewer: ```suggestion
  decided something could be "improved" — refactored, renamed, reworded, or
```

Real real em-dash :)

  decided something could be "improved" - refactored, renamed, reworded, or
  otherwise touched. If you cannot point to a concrete problem you faced as a user,

>   there is no PR to submit.

Reviewer 2: ```suggestion
- **Pull Requests must originate from a real human need.** A valid pull request addresses
  a bug you actually encountered, or a feature you genuinely need. We do not
  accept pull requests where the origin is an AI agent that scanned the codebase and
  decided something could be "improved" -- refactored, renamed, reworded, or
  otherwise touched. If you cannot point to a concrete problem you faced as a user,
  there is no pull request to submit.
```
Same as last one, plus a proper emdash 😅

> where the origin is an AI agent

Just want to highlight that this statement might possibly be revised when security scanning is involved.

>   scanner are welcome, provided the submission still follows all other rules above.

Reviewer: Bold + newline makes this easier to parse, imo

```suggestion
  there is no pull request to submit.\
  **Exception:** Security-related findings from a code scanner are welcome,
  provided the submission still follows all other rules above.
```
